metaquot/ppxlib_metaquot.ml

```diff
@@ -102,11 +154,73 @@ module Expr = Make (struct
 
   let annotate e core_type = pexp_constraint ~loc:core_type.ptyp_loc e core_type
 
-  let cast ext =
+  let fresh_name =
+    let counter = ref 0 in
+    fun () ->
+      let var = "_ppx_metaquot_helper_var%d" ^ string_of_int !counter in
+      incr counter;
+      var
+
+  (* Append the quoted attributes to the attributes present on the
+     antiquoted construct. Take this as example:
+
+     [%expr [%e e] [@attr]]
+
+     Suppose e has pexp_attributes = [attr1]. Then the resulting attributes
+     are [ attr1; [@attr] ]. The decision to put outer attributes (here,
+     [@attr]) at the end of the list is consistent with other parts of ppxlib
+     that accumulate attributes.
+  *)
+  let add_quoted_attributes self e { quoted_attributes; field_name } ~type_name
+      ~loc =
+    match quoted_attributes with
+    | [] -> self#typed e type_name
+    | _ :: _ ->
+        let loc = { loc with loc_ghost = true } in
+        let module Ast_builder_with_loc = (val Ast_builder.make loc) in
```

Comment:
Instantiating all of `Ast_builder` here feels heavyweight. Why not just open `Ast_builder.Default` for the whole file, and pass in `~loc` as necessary?

Reply:
Done.